internal/pipe/reddit/reddit.go

```diff
@@ -30,8 +31,15 @@ func (Pipe) Default(ctx *context.Context) error {
 		ctx.Config.Announce.Reddit.TitleTemplate = defaultTitleTemplate
 	}
 
-	if ctx.Config.Announce.Reddit.URLTemplate == "" {
-		ctx.Config.Announce.Reddit.URLTemplate = defaultURLTemplate
+	switch ctx.TokenType {
+	case context.TokenTypeGitHub:
+		ctx.Config.Announce.Reddit.URLTemplate = defaultGitHubURLTemplate
+	case context.TokenTypeGitLab:
+		ctx.Config.Announce.Reddit.URLTemplate = defaultGitLabURLTemplate
+	case context.TokenTypeGitea:
+		ctx.Config.Announce.Reddit.URLTemplate = defaultGitHubURLTemplate
+	default:
+		return fmt.Errorf("invalid client token type: %q", ctx.TokenType)
```

Comment:
not needed anymore...

```suggestion
	if ctx.Config.Announce.Reddit.URLTemplate == "" {
 		ctx.Config.Announce.Reddit.URLTemplate = defaultURLTemplate
```